File: php/primes.php
<?php

for ($k = 2; $k <200000; $k++)
{
   if (isPrime($k))
   {
       echo 'X';
   }
   else
   {
       echo 'O';
   }
}

function isPrime($number)
{
   for ($i = 2; $i < $number; $i++ )
   {
       if ($number % $i == 0)
       {
           return false;
       }
   }
   return true;
}

File: ruby/primes.rb
def isPrime(number)
  for i in 2..number - 1
		if number % i == 0 then
			return false
		end
	end
        return true
end


for i in 2..200000
	if isPrime(i) then
		print "X"
	else
		print "O"
	end
end
